src/compiler/scala/tools/nsc/typechecker/ContextErrors.scala

```diff
@@ -340,21 +340,48 @@ trait ContextErrors {
 
       //typedSelect
       def NotAMemberError(sel: Tree, qual: Tree, name: Name) = {
-        def errMsg = {
+        import util.{ EditDistance, StringUtil }
+        def errMsg: String = {
           val owner            = qual.tpe.typeSymbol
           val target           = qual.tpe.widen
           def targetKindString = if (owner.isTypeParameterOrSkolem) "type parameter " else ""
           def nameString       = decodeWithKind(name, owner)
           /* Illuminating some common situations and errors a bit further. */
           def addendum         = {
-            val companion = {
+            val companionSymbol: Option[Symbol] = {
               if (name.isTermName && owner.isPackageClass) {
                 target.member(name.toTypeName) match {
-                  case NoSymbol => ""
-                  case sym      => "\nNote: %s exists, but it has no companion object.".format(sym)
+                  case NoSymbol => None
+                  case sym      => Some(sym)
                 }
               }
-              else ""
+              else None
+            }
+            val companion = companionSymbol match {
+              case Some(sym) => "\nnote: %s exists, but it has no companion object.".format(sym)
+              case _         => ""
+            }
+            // find out all the names available under target within 2 edit distances
+            lazy val alternatives: Vector[String] = {
+              val editThreshold = 2
+              val x = name.toString
+              if (x.size < 2) Vector()
+              else {
+                val names = target.nonPrivateMembers.toList.toVector.map(_.name.toString).distinct
+                names filter { n =>
+                  EditDistance.levenshtein(n, x) <= editThreshold
```

Comment:
We should do some pre-filtering, only compute the distance if `math.abs(n.length - x.length) <= 2`.